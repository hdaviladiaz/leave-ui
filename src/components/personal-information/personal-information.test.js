import React from 'react';
import { shallow } from 'enzyme';
import PersonalInformation from './personal-information';
import renderer from 'react-test-renderer';

describe('PersonalInformation', () => {

  const vivi = {"employeeId":"15576",
                "preferredName":"Viviana Perez",
                "picture":{"url":"https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile"},
                "hireDate":"2013-08-05",
                "homeOffice":{"name":"Quito"},
                "workingOffice":{"name":"Pune"}}

  it('should render personal information when person', () => {
    const wrapper = shallow(<PersonalInformation person={vivi} error={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render personal information when error', () => {
    const wrapper = shallow(<PersonalInformation person={null} error={true} />);
    expect(wrapper).toMatchSnapshot();
  });

});
