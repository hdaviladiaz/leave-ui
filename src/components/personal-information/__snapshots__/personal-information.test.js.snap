// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInformation should render personal information when error 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="personal-container col-lg-12"
>
    <Alert
        bsClass="alert"
        bsStyle="danger"
        closeLabel="Close alert"
    >
        Hubo un error al consultar su informaci贸n
    </Alert>
</div>,
  "nodes": Array [
    <div
      className="personal-container col-lg-12"
>
      <Alert
            bsClass="alert"
            bsStyle="danger"
            closeLabel="Close alert"
      >
            Hubo un error al consultar su informaci贸n
      </Alert>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PersonalInformation
        error={true}
        person={null}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "error": true,
          "person": null,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="personal-container col-lg-12"
>
          <Alert
                    bsClass="alert"
                    bsStyle="danger"
                    closeLabel="Close alert"
          >
                    Hubo un error al consultar su informaci贸n
          </Alert>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="personal-container col-lg-12"
>
          <Alert
                    bsClass="alert"
                    bsStyle="danger"
                    closeLabel="Close alert"
          >
                    Hubo un error al consultar su informaci贸n
          </Alert>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PersonalInformation
    error={true}
    person={null}
/>,
}
`;

exports[`PersonalInformation should render personal information when person 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="personal-container col-lg-12"
>
    <div
        className="col-lg-3 no-padding"
    >
        <img
            alt="foto personal"
            className="img-picture"
            src="https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile"
        />
    </div>
    <div
        className="col-lg-9"
    >
        <p>
            Viviana Perez
        </p>
        <p>
            TWer from 
            2013-08-05
        </p>
    </div>
</div>,
  "nodes": Array [
    <div
      className="personal-container col-lg-12"
>
      <div
            className="col-lg-3 no-padding"
      >
            <img
                  alt="foto personal"
                  className="img-picture"
                  src="https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile"
            />
      </div>
      <div
            className="col-lg-9"
      >
            <p>
                  Viviana Perez
            </p>
            <p>
                  TWer from 
                  2013-08-05
            </p>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PersonalInformation
        error={false}
        person={
                Object {
                        "employeeId": "15576",
                        "hireDate": "2013-08-05",
                        "homeOffice": Object {
                          "name": "Quito",
                        },
                        "picture": Object {
                          "url": "https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile",
                        },
                        "preferredName": "Viviana Perez",
                        "workingOffice": Object {
                          "name": "Pune",
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "error": false,
          "person": Object {
            "employeeId": "15576",
            "hireDate": "2013-08-05",
            "homeOffice": Object {
              "name": "Quito",
            },
            "picture": Object {
              "url": "https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile",
            },
            "preferredName": "Viviana Perez",
            "workingOffice": Object {
              "name": "Pune",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="personal-container col-lg-12"
>
          <div
                    className="col-lg-3 no-padding"
          >
                    <img
                              alt="foto personal"
                              className="img-picture"
                              src="https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile"
                    />
          </div>
          <div
                    className="col-lg-9"
          >
                    <p>
                              Viviana Perez
                    </p>
                    <p>
                              TWer from 
                              2013-08-05
                    </p>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="personal-container col-lg-12"
>
          <div
                    className="col-lg-3 no-padding"
          >
                    <img
                              alt="foto personal"
                              className="img-picture"
                              src="https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile"
                    />
          </div>
          <div
                    className="col-lg-9"
          >
                    <p>
                              Viviana Perez
                    </p>
                    <p>
                              TWer from 
                              2013-08-05
                    </p>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PersonalInformation
    error={false}
    person={
        Object {
            "employeeId": "15576",
            "hireDate": "2013-08-05",
            "homeOffice": Object {
              "name": "Quito",
            },
            "picture": Object {
              "url": "https://jigsaw.thoughtworks.net/consultants/22118/show_picture?style=profile",
            },
            "preferredName": "Viviana Perez",
            "workingOffice": Object {
              "name": "Pune",
            },
          }
    }
/>,
}
`;
